weave

```diff
@@ -986,12 +986,15 @@ dns_args() {
 }
 
 etc_hosts_contents() {
-    NAME=$1
+    FQDN=$1
     shift
+    NAME=${FQDN%%.*}
+    HOSTNAMES=$([ "$NAME" = "$FQDN" -o "$NAME." = "$FQDN" ] && echo "$NAME" || echo "$FQDN $NAME")
```

Comment:
We should make the logic here and in `with_container_fqdn` consistent. I kinda prefer this version since it doesn't require grep.

Don't like the sub-shell though.

```
NAME=${FQDN%%.*}
HOSTNAMES="$NAME"
[ "$NAME" = "$FQDN" -o "$NAME." = "$FQDN" ] || HOSTNAMES="$FQDN $HOSTNAMES"
```
should work.